### ISSUE TITLE
Better defaults for file alt text

### ISSUE THREAD

**Issue by brynwhyman** [silverstripe/silverstripe-asset-admin#983]
# Overview
We're combining a couple of issues relating to improving the default handling of alt text for files:
* [Alt text defaults to the image's title if not set on CMS](https://github.com/silverstripe/silverstripe-framework/issues/9099) from @Gazzayeatman
* [When inserting images to content, alt text should autofill
](https://github.com/silverstripe/silverstripe-asset-admin/issues/840) (similar to)

We need to ensure that the user is guided to make the right decisions here and is not unknowingly adding alt text that will not be beneficial to screen readers.

# Two changes:

#### 1. If alt text is empty the `alt` attribute should be empty

At the moment an image's alt text defaults to the image's title if there is no alt text set. There are some cases where the user does no want alt text at all. See: https://www.w3.org/WAI/tutorials/images/decorative/

Note, this isn't reflected on the admin UI, but can be seen in the image HTML.

#### 2. Guide user to add alt text for files

There should be an improvement in prompting users that alt text is probably required for a file (while keeping the use case from the above change in mind).

Suggestions:
- There is clear messaging advising users to add alt text for the benefit of screen readers
  - Noting that in CWP 1.x there was a prompt when the alt text is left empty: https://github.com/silverstripe/cwp/issues/149
- Do not auto-populate the alt text field to avoid setting alt text like "image 1"
- Note: this should only be implemented for the WYSIWYG, not for the upload field

#### ~(Stretch goal) 3. Reorder the fields presented for placing a file~
Tracking in: https://github.com/silverstripe/silverstripe-asset-admin/issues/991

### PRs

- [x] https://github.com/silverstripe/silverstripe-admin/pull/964
- [x] https://github.com/silverstripe/silverstripe-framework/pull/9281
- [ ] https://github.com/silverstripe/silverstripe-asset-admin/pull/1027

**Comment by brynwhyman** [comment]
CC @silverstripeux, I've combined a number of suggestions

**Comment by brynwhyman** [comment]
@silverstripeux would be great to have some input on the alert pattern

**Comment by clarkepaul** [comment]
@brynwhyman I've added a third item to this list which might affect the point size. It's to order the ALT field so that it's after the caption field, also raised as a separate issue.

**Comment by brynwhyman** [comment]
Thanks @clarkepaul, I've added it as a stretch goal for this story

**Comment by clarkepaul** [comment]
Design mockups for discussion... append an action to the end of the ALT field which triggers a popover for more information about how/when to use ALT text, it's not enforced but recommended. We could open the popover automatically when the field is focused if we want to push the recommendation a bit more https://projects.invisionapp.com/dsm/silver-stripe/silver-stripe/asset/components/5d70ab0cd81b4b445753d31b

I'm hoping this mix of components would work out of the box but not completely sure.

**Comment by brynwhyman** [comment]
Looks like a pretty similar discussion happening on https://github.com/silverstripe/silverstripe-assets/issues/129

**Comment by mr45144** [comment]
pointed here from https://github.com/silverstripe/silverstripe-assets/issues/129 by @brynwhyman Thanks for that. I'm happy that this covers the majority of my discussion over there.

Two concerns:
1. Just for clarity an empty ALT text is ALT="" not ALT  
2. There is mention above of this being only in the WYSIWYG and not the uploader - can we clarify that this would be applied when adding an image to a page banner (ie straight to the uploadr) and to data objects (correct term?) like section, gallery, timeline, notice. Basically the content owner needs to be able to add a contextually valid alt text whenever they add an image to a front end facing context. Hopefully I'm confusing the use of uploader but you can upload directly in TinyMCE or therough a page/object.

**Comment by brynwhyman** [comment]
Hi @mr45144, agree that an image without alt text should have an attribute of `alt=''`.

>There is mention above of this being only in the WYSIWYG and not the uploader

At this stage, there is no path for a user to add alt text for an image uploaded through the uploadfield commonly used for data objects. This is however possible when adding an image through the WYSIWYG, see the following screenshot:
![image](https://user-images.githubusercontent.com/15868440/65288116-c7330680-db99-11e9-9fc4-e1f88bc3ba60.png)

Our intention is to make improvements for the WYSIWYG flow and then look at copying that functionality over for use with the uploadfield, so when relevant content editors can added alt text and tooltips.

**Comment by mr45144** [comment]
I support that intention - as you say, even after tying yourself in knots to get to the original file on a banner image you are not even presented with an ALT text field.

**Comment by maxime-rainville** [comment]
I peer reviewed the admin and framework PR. There wasn't any change to the `alt` field. I presumed you're waiting for the generic TIP PRs to get over the line before updating the actual form field.

**Comment by Cheddam** [comment]
Final PR to close this out is available at #1027.

**Comment by bigweb-ai-agency** [comment]
Hi all, we have a situation where our content guys created more than 1000 content  pages filled with 3-10 images per page. All of images are now missing ALT , as they switched form SS3 where ALT was autoadded and all images have nicley formed filenames and titles. Could you please advise on which would be the best way of fixing this, so that frontend renders alt as image title ? Also maybe it is possible to reenable alt autofill somehow in config ? Thank you.
